app/utils/auth.server.ts

```diff
@@ -255,3 +256,33 @@ export async function verifyUserPassword(
 
 	return { id: userWithPassword.id }
 }
+
+export async function checkCommonPassword(password: string) {
+	const hash = crypto
+		.createHash('sha1')
+		.update(password, 'utf8')
+		.digest('hex')
+		.toUpperCase()
+
+	const [prefix, suffix] = [hash.slice(0, 5), hash.slice(5)]
+
+	const controller = new AbortController()
+
+	try {
+		const timeoutId = setTimeout(() => controller.abort(), 1000)
+
+		const res = await fetch(`https://api.pwnedpasswords.com/range/${prefix}`)
+
+		clearTimeout(timeoutId)
+
+		if (!res.ok) false
+
+		const data = await res.text()
+		return data.split('/\r?\n/').some((line) => line.includes(suffix))
+	} catch (error) {
+		if (error instanceof DOMException && error.name === 'AbortError') {
```

Comment:
In Node.js the error thrown is an `AbortError` not a `DOMException`. Could you update this? Could probably just do this:

```suggestion
		if (error instanceof Error && error.name === 'AbortError') {
```

Reply:
actually, it was copilot mistake, and I didn't paid attention to it sorry for that